app/javascript/features/lobby/lobbySlice.js

```diff
@@ -66,15 +66,51 @@ export const leaveBattle = createAsyncThunk(
 )
 
 export const readyPlayer = createAsyncThunk(
-  "lobby/readyPlayer",
-  async ({ battleId, playerId }, {_, rejectWithValue}) => {
-    try {
-      const response = await axios.post("/battles/player/ready", { battleId, playerId })
-      return response.data
-    } catch (err) {
-      return rejectWithValue(err.response.data)
+    "lobby/readyPlayer",
+    async ({ battleId, playerId }, {_, rejectWithValue}) => {
+        try {
+            const response = await axios.post("/battles/player/ready", { battleId, playerId })
+            return response.data
+        } catch (err) {
+            return rejectWithValue(err.response.data)
+        }
+    }
+)
+
+export const unreadyPlayer = createAsyncThunk(
+    "lobby/unreadyPlayer",
+    async ({ battleId, playerId }) => {
+        try {
+            const response = await axios.post("/battles/player/unready", { battleId, playerId })
+            return response.data
+        } catch (err) {
+            return rejectWithValue(err.response.data)
+        }
+    }
+)
+
+export const abandonPlayer = createAsyncThunk(
+    "lobby/abandonPlayer",
+    async ({ battleId, playerId }) => {
+        try {
+            const response = await axios.post("/battles/player/abandon", { battleId, playerId })
+            return response.data
+        } catch (err) {
+            return rejectWithValue(err.response.data)
+        }
+    }
+)
+
+export const unabandonPlayer = createAsyncThunk(
+    "lobby/unabandonPlayer",
+    async ({ battleId, playerId }) => {
+        try {
+            const response = await axios.post("/battles/player/unabandon", { battleId, playerId })
+            return response.data
+        } catch (err) {
+            return rejectWithValue(err.response.data)
+        }
     }
-  }
 )
 
```

Comment:
Need to add reducers cases for these requests, ex:
```
      .addCase(abandonBattle.pending, (state, action) => {
        state.errorMessage = null
        state.isPending = true
      })
      .addCase(abandonBattle.fulfilled, (state, action) => {
        state.errorMessage = null
        state.isPending = false
      })
      .addCase(abandonBattle.rejected, (state, action) => {
        state.errorMessage = action.payload.error
        state.isPending = false
      })
```
This allows us to both disable buttons when a request is pending and also surface an error alert if the request fails